src/decorators.js

```diff
@@ -21,3 +25,34 @@ export function on(...predicate) {
     target.events.push([predicate, target[property]]);
   };
 }
+
+
+/**
+ * Decorator used with class-based React components.
+ * It provides all the required props and helpers for tanok internals.
+ *
+ * Usage example:
+ *
+ * @tanokComponent
+ * class MyComponent extends React.Component {
+ *    ... your component methods.
+ * }
+ *
+ * */
+export function tanokComponent(target) {
```

Comment:
Can't see a difference with HOC in `src/component.js`. Except that HOC version first creates a subclass, which isn't necessary as it turns out.

I suggest that we use our existing HOC as decorator. So I ask you to remove subclassing from HOC and copy your comment with usage example there.

The rest is absolutely relevant, of course.